Realm.Shared/Realm.cs

```diff
@@ -194,12 +194,22 @@ private RealmObject.Metadata CreateRealmObjectMetadata(Schema.ObjectSchema schem
                 helper = Dynamic.DynamicRealmObjectHelper.Instance;
             }
 
+            var colMap = new Dictionary<string, IntPtr>();
```

Comment:
Do we need to shorten those? Even though it's kind of obvious what it means, I find full names more readable. Additionally, it's inconsistent with other names, e.g. `Metadata.ColumnIndices`.